### DISCUSSION TITLE
How to highlight the part of text?

### DISCUSSION THREAD

**Discussion by ShadowmourneE** [MurhafSousli/ngx-highlightjs#197] [Q&A] [answered]
I use it in this way
```
<pre>
<code [highlight]="MYTEXT" [lineNumbers]="true" (highlighted)="onHighlighted($event)" >
</code>
</pre>
```

my text is like:

 'text text text `<span class="myCustomClass">**TEXT BLOCK TO HIGHLIGHT**</span>'`
 
 Bu result will be the same(with span):` 'text text text <span class="myCustomClass">TEXT BLOCK TO HIGHLIGHT</span>'`
 
 How can I apply this class and span in html to the rest(class="hljs-attr" etc.) and not parse it as separate text:
 I mean in order to result should be:

  'text text text `**_TEXT BLOCK TO HIGHLIGHT WITH CUSTOM STYLES_**'`
  
  I found the method fixMarkup but I am not sure how to use it correctly... I assume that there is a way to change the existing markup onHighlight event when it processed, but in this way it is impossible to normal even replace it because the result html would be like:
  
  
  `**span**</span> <span class="hljs-attr">**class**</span>=<span class="hljs-string">**myCustomClass&quo** ..... etc.`
  
  I would like to find the appropriate way how to use it... thanks

**Comment by MurhafSousli** [comment, marked as answer]
Honestly, I don't know what fixMarkup does, you can ask on their main repo regarding the special cases, this is just a wrapper plugin and the component is meant for highlighting a block of text
